azure-kusto-ingest/azure/kusto/ingest/_ingestion_properties.py

```diff
@@ -112,13 +122,25 @@ def __init__(
         validationPolicy=None,
         additionalProperties=None,
     ):
-        if mapping is not None and mappingReference is not None:
+        # mapping_reference will be deprecated in the following versions
+        if mappingReference is not None:
+            warnings.warn(
+                "mappingReference will be deprecated in the following versions."
+                " Please use ingestionMappingReference instead",
+                PendingDeprecationWarning,
+            )
+        if (mapping is not None and (mappingReference is not None or ingestionMappingReference is not None)) or (
```

Comment:
this is hard to read (and i think is wring as well, considering you have the same condition twice).
break it down into multiple, logically distinguishable ifs (if needed, use vars to make it readable , like `mapping_exists = mapping is not None or mappingReference is not None`)